src/app/content/search/hooks/index.ts

```diff
@@ -36,86 +38,89 @@ export const receiveSearchHook: ActionHookBody<typeof receiveSearchResults> = (s
   const targetPage = assertDefined(
     findArchiveTreeNodeById(book.tree, targetPageId),
     'search result pointed to page that wasn\'t in book'
   );
-
-  const savedQuery = savedSearch ? savedSearch.query : null;
-  if (
-    savedQuery === query &&
-    page.id === stripIdVersion(targetPage.id) &&
-    isEqual(select.selectedResult(state), selectedResult)
-  ) {
-    return; // if search and page match current history record, noop
-  }
-
-  if (book.id !== getIndexData(selectedResult.result.index).bookId) {
-    return;
-  }
-
-  services.dispatch(selectSearchResult(selectedResult));
-
   const navigation = {
     params: getBookPageUrlAndParams(book, targetPage).params,
     route: content,
     state : {
       bookUid: book.id,
       bookVersion: book.version,
       pageUid: stripIdVersion(targetPage.id),
-      search: {query, selectedResult},
     },
   };
 
   const action = stripIdVersion(page.id) === stripIdVersion(targetPage.id) ? replace : push;
+  const search = queryString.stringify({
+    query,
+    target: JSON.stringify({ type: 'search', index: selectedResult.highlight }),
+  });
+  const hash = selectedResult.result.source.elementId;
 
-  services.dispatch(action(navigation));
+  services.dispatch(action(navigation, { hash, search }));
 };
 
 export const clearSearchHook: ActionHookBody<typeof clearSearch | typeof openToc> = (services) => () => {
-  if (services.history.location.state && services.history.location.state.search) {
+  const scrollTarget = selectNavigation.scrollTarget(services.getState());
+  if (scrollTarget && isSearchScrollTarget(scrollTarget)) {
     services.history.replace({
-      state: {
-        ...services.history.location.state,
-        search: null,
-      },
+      hash: '',
+      search: '',
     });
   }
 };
 
 // composed in /content/locationChange hook because it needs to happen after book load
-export const syncSearch: RouteHookBody<typeof content> = (services) => async(locationChange) => {
-  const query = select.query(services.getState());
-  const selectedResult = select.selectedResult(services.getState());
-  const savedSearch = getSearchFromLocation(locationChange.location);
-
-  if (savedSearch && savedSearch.query && savedSearch.query !== query) {
-    services.dispatch(
-      requestSearch(
-        savedSearch.query,
-        savedSearch.selectedResult ? {
-          isResultReload: true,
-          selectedResult: savedSearch.selectedResult,
-        } : undefined
-      )
-    );
-  } else if (savedSearch && savedSearch.selectedResult && !isEqual(savedSearch.selectedResult, selectedResult)) {
-    services.dispatch(selectSearchResult(savedSearch.selectedResult));
-  } else if ((!savedSearch || !savedSearch.query) && query) {
-    services.dispatch(clearSearch());
+export const syncSearch: RouteHookBody<typeof content> = (services) => async() => {
+  const state = services.getState();
+  const navigationQuery = selectNavigation.localState(state).query.query;
```

Comment:
There is a selector that selects the `query` but it's also is called `query` so you end up with the same amount of `query`. Not sure which one is better.